### PR TITLE
[master] Use `toJson` method from data.jsondata module in http client

### PR DESCRIPTION
**Pull request by lnash94**
## Purpose
Fix: https://github.com/ballerina-platform/ballerina-library/issues/7012

Client data binding integration: https://github.com/ballerina-platform/ballerina-library/issues/6747
Service data binding integration: https://github.com/ballerina-platform/ballerina-library/issues/7011

Disable the below test because of data.json data module currently does not support the `table` types
and enable tests once this issue is done: https://github.com/ballerina-platform/ballerina-library/issues/7020

- testIntTableDatabinding()
- testIntTableOrXmlArrayDatabinding()
- testIntTableDatabindingByType()
- testStringTableDatabinding()
- testStringTableDatabindingByType()
- testRecordTableDatabinding()
- testRecordTableDatabindingByType()
- testByteArrTableDatabinding()
- testByteArrTableDatabindingByType()

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests
- [ ] Updated the spec
- [ ] Checked native-image compatibility
- [ ] Checked the impact on OpenAPI generation

Dependent PR : https://github.com/ballerina-platform/module-ballerina-data.jsondata/pull/35 once this is merged , test can be added with new data.jsondata timestamp

### CODE REVIEW

**Comment by lnash94** [comment]
Checking the feasibility of enabling this implementation because we encounter this issue from jsondata module https://github.com/ballerina-platform/ballerina-library/issues/7008

File: ballerina/http_request.bal [hunk 1]
```diff
@@ -541,7 +544,7 @@ public class Request {
         } else if payload is mime:Entity[] {
             self.setBodyParts(payload);
         } else if payload is anydata {
-            self.setJsonPayload(payload.toJson());
+            self.setJsonPayload(jsondata:toJson(payload));
```
Comment (TharmiganK) [inline ballerina/http_request.bal:547]: Shall we have a separate function for anydata values? Using `setJsonPayload` does the conversion again which is redundant.

```ballerina
isolated function setAnydataAsJsonPayload(anydata payload) {
     mime:Entity entity = self.getEntityWithoutBodyAndHeaders();
     setJson(entity, jsondata:toJson(payload), self.getContentType(), contentType);
     self.setEntityAndUpdateContentTypeHeader(entity);
}
```

File: ballerina/http_response.bal [hunk 2]
```diff
@@ -521,7 +522,7 @@ public class Response {
         } else if payload is stream<SseEvent, error?> {
             self.setSseEventStream(payload);
         } else if payload is anydata {
-            self.setJsonPayload(payload.toJson());
+            self.setJsonPayload(jsondata:toJson(payload));
```
Comment (TharmiganK) [inline ballerina/http_response.bal:525]: Same as the request method. Have a separate function for setting anydata payload as json payload

File: gradle.properties [hunk 3]
```diff
@@ -25,6 +25,9 @@ protobufVersion=3.20.3
 jacocoVersion=0.8.10
 ballerinaToOpenApiVersion=2.1.0
 swaggerCoreVersion=2.2.22
+javaJsonPathVersion=2.9.0
```
Comment (TharmiganK) [inline gradle.properties:28]: I think we added these to fix the compiler failure of jsondata. Can we remove them and check? IIRC these are only needed if we have used them in the native code

File: native/src/main/java/io/ballerina/stdlib/http/api/nativeimpl/ExternResponseProcessor.java [hunk 4]
```diff
@@ -296,6 +297,14 @@ private static Object getMediaType(BObject response, Type mediaTypeType, boolean
             return null;
         }
         try {
+//            Map<String, Object> valueMap = new HashMap<>();
```
Comment (TharmiganK) [inline native/src/main/java/io/ballerina/stdlib/http/api/nativeimpl/ExternResponseProcessor.java:300]: We can remove these commented lines

File: ballerina/http_request.bal (hunk 1 above)
Comment (lnash94) [reply ballerina/http_request.bal:547]: Yeah , I will make it separate, then are we going to use `contentType` value as default value ?

File: gradle.properties (hunk 3 above)
Comment (lnash94) [reply gradle.properties]: Yeah, we added these to check the jsondata compiler plugin issue and we can remove this.

**Comment by lnash94** [comment]
Added test fail due to the https://github.com/ballerina-platform/ballerina-library/issues/7024

**Pull request closed without merging**